src/AWS.Deploy.ServerMode.Client/ServerModeSession.cs

```diff
@@ -0,0 +1,260 @@
+                var startServerTask = RunCommand(command, keyInfoStdin);
+
+                _baseUrl = $"http://localhost:{port}";
+                var isServerAvailableTask = IsServerAvailable(_baseUrl, cancellationToken);
+
+                if (isServerAvailableTask == await Task.WhenAny(startServerTask, isServerAvailableTask).ConfigureAwait(false))
+                {
+                    // The server timed out, this isn't a transient error, therefore, we throw
+                    if (!isServerAvailableTask.Result)
+                    {
+                        throw new InternalServerModeException($"\"{command}\" failed for unknown reason.");
+                    }
+
+                    // Server has started, it is safe to return
+                    return;
+                }
+
+                // For -100 errors, we want to check all the ports in the configured port range
+                // If the error code other than -100, this is an unexpected exit code.
+                if (startServerTask.Result != TCP_PORT_ERROR)
+                {
+                    throw new InternalServerModeException($"\"{command}\" failed for unknown reason.");
+                }
+            }
+
+            throw new PortUnavailableException($"Free port unavailable in {_startPort}-{_endPort} range.");
+        }
+
+        public bool TryGetRestAPIClient(Func<Task<AWSCredentials>> credentialsGenerator, out IRestAPIClient? restApiClient)
+        {
+            if (_baseUrl == null || _aes == null)
+            {
+                restApiClient = null;
+                return false;
+            }
+
+            var httpClient = ServerModeHttpClientFactory.ConstructHttpClient(credentialsGenerator, _aes);
+            restApiClient = new RestAPIClient(_baseUrl, httpClient);
+            return true;
+        }
+
+        public bool TryGetDeploymentCommunicationClient(out IDeploymentCommunicationClient? deploymentCommunicationClient)
+        {
+            if (_baseUrl == null || _aes == null)
+            {
+                deploymentCommunicationClient = null;
+                return false;
+            }
+
+            deploymentCommunicationClient = new DeploymentCommunicationClient(_baseUrl);
+            return true;
+        }
+
+        public async ValueTask<bool> IsAlive(CancellationToken cancellationToken)
+        {
+            if (_baseUrl == null)
+            {
+                return false;
+            }
+
+            try
+            {
+                return await IsServerAvailable(_baseUrl, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        #region Private methods
+
+        private Task<bool> IsServerAvailable(string baseUrl, CancellationToken cancellationToken)
+        {
+            var healthUrl = $"{baseUrl}/api/v1/health";
```

Comment:
Is there any way to avoid hardcoding and keep this in sync with the endpoint url?